invoke.go

```diff
@@ -0,0 +1,178 @@
+package function
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"github.com/google/uuid"
+)
+
+const (
+	DefaultInvokeSource      = "/boson/fn"
+	DefaultInvokeType        = "boson.fn"
+	DefaultInvokeContentType = "text/plain"
+	DefaultInvokeData        = "Hello World"
+	DefaultInvokeFormat      = "http"
+)
+
+// InvokeMesage is the message used by the convenience method Invoke to provide
+// a simple way to trigger the execution of a Function during development.
+type InvokeMessage struct {
+	ID          string
+	Source      string
+	Type        string
+	ContentType string
+	Data        string
+	Format      string //optional override for Function-defined message format
+}
+
+// NewInvokeMessage creates a new InvokeMessage with fields populated
+func NewInvokeMessage() InvokeMessage {
+	return InvokeMessage{
+		ID:          uuid.NewString(),
+		Source:      DefaultInvokeSource,
+		Type:        DefaultInvokeType,
+		ContentType: DefaultInvokeContentType,
+		Data:        DefaultInvokeData,
+		// Format override not set by default: value from Function being preferred.
+	}
+}
+
+// invoke the Function instance in the target environment with the
+// invocation message.
+func invoke(ctx context.Context, c *Client, f Function, target string, m InvokeMessage) error {
+
+	// Get the first available route from 'local', 'remote', a named environment
+	// or treat target
+	route, err := invocationRoute(ctx, c, f, target) // choose instance to invoke
+	if err != nil {
+		return err
+	}
+
+	// Format" either 'http' or 'cloudevent'
+	// TODO: discuss if providing a Format on Message should a) update the
+	// Function to use the new format if none is defined already (backwards
+	// compatibility fix) or b) always update the Function, even if it was already
+	// set. Once decided, codify in a test.
```

Comment:
We could ask the user if they want this value saved, iff func.yaml does not contain it or differs.

```
You provided the `http` as the format for this Function.
Would you like to save this for future invocations?
```

Reply:
I like the idea of prompting.
The conundrum for me was: what should we choose if the process is not being run in an interactive terminal, just update, making the choice by default "sticky"?  That seems like not a good idea, because side-effects.  My inital thought was that the most straightforward and documentation-friendly option might be to add a `--save` flag to `invoke` which would state "save the given format for future invocations".  This could perhaps be the value being prompted for by your suggestion (if not provided of course).

I will add this to my "punts" list now...